src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -84,9 +84,19 @@ const mongoSchemaFromFieldsAndClassNameAndCLP = (
   };
 
   for (const fieldName in fields) {
+    const { type, targetClass, ...fieldOptions } = fields[fieldName];
     mongoObject[
       fieldName
-    ] = MongoSchemaCollection.parseFieldTypeToMongoFieldType(fields[fieldName]);
+    ] = MongoSchemaCollection.parseFieldTypeToMongoFieldType({
+      type,
+      targetClass,
+    });
+    if (fieldOptions && Object.keys(fieldOptions).length > 0) {
```

Comment:
Can you add a test for setting required fields and defaults when you create a new class / new class schema?

Reply:
done
https://github.com/parse-community/parse-server/pull/5835/files#diff-062f88bd150cc9bdbefd675f4267241cR409